x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/execution/search/extractor/MetricAggExtractor.java

```diff
@@ -41,22 +48,31 @@
     private final String name;
     private final String property;
     private final String innerKey;
-    private final boolean isDateTimeBased;
+    private final DataType dataType;
     private final ZoneId zoneId;
 
-    public MetricAggExtractor(String name, String property, String innerKey, ZoneId zoneId, boolean isDateTimeBased) {
+    public MetricAggExtractor(String name, String property, String innerKey, ZoneId zoneId, @Nullable DataType dataType) {
```

Comment:
Nit: zoneId is declared before dataType (ok) but the assignment occurs before - the code was like that beforehand but let's improve it now.
Second nit : if metricAggExtractor is called a lot with no DataType, it's worth looking at declaring a separate constructor to perform delegation.

Reply:
Addressed.